go-controller/pkg/ovn/egressip.go

```diff
@@ -605,10 +615,95 @@ func (oc *Controller) reconcileCloudPrivateIPConfig(old, new *ocpcloudnetworkapi
 				return err
 			}
 		}
+
+		oc.eIPC.pendingCloudPrivateIPConfigsMutex.Lock()
+		defer oc.eIPC.pendingCloudPrivateIPConfigsMutex.Unlock()
+		// Remove the finished add / update operation from the pending cache. We
+		// never process add and deletes in the same sync, and for updates:
+		// deletes are always performed before adds, hence we should only ever
+		// fully delete the item from the pending cache once the add has
+		// finished.
+		ops, pending := oc.eIPC.pendingCloudPrivateIPConfigsOps[egressIPName]
+		if !pending {
+			// Do not return an error here, it will lead to spurious error
+			// messages on restart because we will process a bunch of adds for
+			// all existing objects, for which no CR was issued.
+			klog.V(5).Infof("No pending operation found for EgressIP: %s while processing created CloudPrivateIPConfig", egressIPName)
+			return nil
+		}
+		op, exists := ops[egressIPString]
+		if !exists {
+			klog.V(5).Infof("Pending operations found for EgressIP: %s, but not for the created CloudPrivateIPConfig: %s", egressIPName, egressIPString)
+			return nil
+		}
+		if (op.toAdd != "" && op.toDelete != "") || (op.toAdd != "" && op.toDelete == "") {
```

Comment:
isn't it the same as `if op.toAdd != ""`?

Reply:
It is. I will update it